src/Microsoft.Health.Dicom.SqlServer/Features/Store/SqlIndexDataStore.cs

```diff
@@ -87,6 +83,50 @@ public async Task<long> CreateInstanceIndexAsync(DicomDataset instance, Cancella
             }
         }
 
+        private void PopulateAddInstanceCommand(SqlCommandWrapper wrapper, DicomDataset instance, IEnumerable<IndexTag> indexTags)
+        {
+            if (_schemaInformation.Current >= SchemaVersionConstants.SupportCustomTagSchemaVersion)
```

Comment:
Is there a better pattern than this? Seems non-managable long term.

Think of it like this...

Each n.sql needs different store service.

SqlIndexDataStore : IIndexDataStore
virtual CreateInstanceIndexAsync()
static SqlIndexDataStore GetStoreForSchemaVersion(IEnumberable<IIndexDataStore>)
{
    switch(Schemaversion)
    ...
}

SqlIndexDataStore1 : SqlIndexDataStore
override CreateInstanceIndexAsync()
{ version 1 method here }

SqlIndexDataStore2 : SqlIndexDataStore1
override CreateInstanceIndexAsync()
{ version 2 method here}

Others methods are inherited and work as before.

Now in ServiceClass, you take Ienumerable of SqlIndexStore 
and call SqlIndexDataStore.GetStoreForSchemaVersion()
and use that to call the store APIs.

Reply:
I have a different way to implement this, please take a look.